Task Controller and README updates

jasonjewik opened the pull request:
1. Added .dockerignore so that `make build` wouldn't go through the node modules and take forever on a Windows machine.
2. Added GET and POST requests for tasks to README.md
3. Changed task controller so that API responses contain both "data" and "error".

**POST** /api/tasks/

![image](https://user-images.githubusercontent.com/13477221/68883479-00818180-06c6-11ea-9339-22d5b7fb1ee9.png)

**POST** /api/tasks/ (bad request - missing required 'name' field)

![image](https://user-images.githubusercontent.com/13477221/68883604-3aeb1e80-06c6-11ea-9837-039adc154aad.png)

**GET** /api/tasks/{id}

![image](https://user-images.githubusercontent.com/13477221/68883533-168f4200-06c6-11ea-95e9-3a0fc8f75b07.png)

alexlongerbeam commented on README.md:
```diff
@@ -22,20 +22,57 @@ Use the `access_token` from Chrome to do requests
+      "created_at": <CREATED_AT_TIMESTAMP>
+	},
+	"error": "<ERROR_MESSAGE>",
+}
```
Why these changes? The Readme formatting looks kinda weird for this now

alexlongerbeam replied:
Maybe remove one tab indent from all the data fields in the original version but I think you removed both

jasonjewik replied:
Oh, okay, I see what you mean. For some reason it rendered differently on my computer when I was working on it earlier, but I'll definitely clean it up.

alexlongerbeam commented on README.md:
````diff
@@ -22,20 +22,57 @@ Use the `access_token` from Chrome to do requests
+    "error": "<ERROR_MESSAGE>"
+}
+```
+
````
Can we actually just return the whole task here? So this is a README and API change. (Also fix formatting here)

alexlongerbeam commented on src/controllers/task.js:
```diff
@@ -33,49 +12,31 @@ const TaskController = (taskModel) => {
         const name = body.name;
         const description = body.description;
         const duration = body.duration;
         const [data, err] = await taskModel.createTask(name, description, duration);
```
Instead, just check if "name" or "duration" is undefined after getting it from the body (duration should definitely be a required field)

alexlongerbeam commented on src/controllers/task.js:
```diff
@@ -33,49 +12,31 @@ const TaskController = (taskModel) => {
         if (!req.body)
             return res.status(400).json({
-                message: "Malformed Request"
+                error: "Malformed Request"
```
For this return as well as the error return below, can you also add an empty "data" field? Makes things more consistent if every API return has both data and error, regardless of what it is

jasonjewik commented:
1. Fixed formatting issues in README.md
2. Changed all API return messages to include `data` and `error` fields.
3. Changed checks for undefined fields to happen after assignment instead of before.

**POST** /api/tasks/
![image](https://user-images.githubusercontent.com/13477221/68974533-d7382280-07a5-11ea-8bec-8111bd5b2351.png)

**GET** /api/tasks/{id}
![image](https://user-images.githubusercontent.com/13477221/68974546-e28b4e00-07a5-11ea-9818-e8d820c0c2eb.png)

**POST** /api/events/
![image](https://user-images.githubusercontent.com/13477221/68974607-0189e000-07a6-11ea-907b-7cf7bcde0c5f.png)

**GET** /api/events/{id}
![image](https://user-images.githubusercontent.com/13477221/68974628-0e0e3880-07a6-11ea-8679-8c26c27dc949.png)

alexlongerbeam commented on README.md:
```diff
@@ -22,20 +22,96 @@ Use the `access_token` from Chrome to do requests
+{
+  "name": "<event_name>"
+}
+
```
Missing an extra ``` here, the rest of the readme got formatted backwards
